Makefile

```diff
@@ -0,0 +1,117 @@
+# Based on https://gist.github.com/trosendal/d4646812a43920bfe94e
+
+DEPTAG := 1.0.7
+DEPLIBNAME := ed25519_bip32
+DEPLOC := https://github.com/spacemeshos/$(DEPLIBNAME)/releases/download
+DEPLIB := lib$(DEPLIBNAME)
+# Exclude dylib files (we only need the static libs)
+EXCLUDE_PATTERN := "LICENSE" "*.so" "*.dylib"
+UNZIP_DEST := deps
+REAL_DEST := $(shell realpath .)/$(UNZIP_DEST)
+DOWNLOAD_DEST := $(UNZIP_DEST)/$(DEPLIB).tar.gz
+EXTLDFLAGS := -L$(UNZIP_DEST) -l$(DEPLIBNAME)
+
+# Detect operating system
+ifeq ($(OS),Windows_NT)
+  SYSTEM := windows
+else
+  UNAME_S := $(shell uname -s)
+  ifeq ($(UNAME_S),Linux)
+	SYSTEM := linux
+  else ifeq ($(UNAME_S),Darwin)
+	SYSTEM := darwin
+  else
+	$(error Unknown operating system: $(UNAME_S))
+  endif
+endif
+
+# Default values. Can be overridden on command line, e.g., inside CLI for cross-compilation.
+# Note: this Makefile structure theoretically supports cross-compilation using GOOS and GOARCH.
+# In practice, however, depending on the host and target OS/architecture, you'll likely run into
+# errors in both the compiler and the linker when trying to compile cross-platform.
+GOOS ?= $(SYSTEM)
+GOARCH ?= unknown
```

Comment:
Instead of overwriting these; `GOOS` and `GOARCH` are actually pretty useful in figuring whats the current operating system and architecture.

I think we should not try to make cross compilation possible with our `Makefile` (as soon as `cgo` is involved this becomes a near impossible task anyway). Then we can use `GOOS` and `GOARCH` to identify on which platform we are instead of identifying the platform and setting them ourselves.

Reply:
I hear ya, and I agree with your point that cross compilation is basically impossible anyway, but I don't see an advantage to using `GOOS` and `GOARCH` here versus the way this is written (`OS`, `uname` etc.)